test/shared/src/main/scala/zio/test/RunnableSpec.scala

```diff
@@ -44,16 +44,23 @@ abstract class RunnableSpec[R, E] extends AbstractRunnableSpec {
   /**
    * A simple main function that can be used to run the spec.
    */
-  override final def run: ZIO[ZEnv with Has[ZIOAppArgs], Any, Any] =
-    for {
-      args     <- ZIO.service[ZIOAppArgs]
-      testArgs  = TestArgs.parse(args.args.toArray)
-      exitCode <- run(spec, testArgs).provideLayer(runner.bootstrap)
-      _        <- doExit(exitCode)
-    } yield ()
+  final def main(args: Array[String]): Unit = {
```

Comment:
This seems a regression of sorts, since you now have to implement `main` rather than `ZIOApp#run`?

Reply:
This is the old `RunnableSpec` which isn't constrained to the environment being a subtype of `Has` so I reverted to where this does not extend `ZIOApp`. I think this is okay because going forward we want to use `ZIOSpec` for everything, which does extend `ZIOApp`, though we have to finish the word there of implementing support for that in the SBT test runner.